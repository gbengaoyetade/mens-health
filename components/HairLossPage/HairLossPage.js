import { useState } from 'react';
import style from './hairlosspage.module.css';
import Image from 'next/image';
import Quiz from '../Quiz';

const HairLossPage = ({ questions }) => {
  const [showQuiz, setShowQuiz] = useState(false);

  const handleTakeQuiz = () => {
    setShowQuiz(true);
  };

  if (showQuiz) {
    return (
      <Quiz
        showQuiz={showQuiz}
        setShowQuiz={setShowQuiz}
        questions={questions}
      />
    );
  }

  return (
    <div className={style.main}>
      <main>
        <section className={style['hero-section']}>
          <div>
            <Image
              alt='app logo'
              src='/images/symbol.svg'
              width='40px'
              height='38px'
            />
          </div>

          <div className={style['head-text']}>
            <h1>Be good to yourself</h1>
            <p className={style['hero-paragraph']}>
              We’re working around the clock to bring you a holistic approach to
              your wellness. From top to bottom, inside and out.
            </p>
            <button onClick={handleTakeQuiz} className={style.btn}>
              TAKE THE QUIZ
            </button>
          </div>
        </section>

        <section className={style['help-section']}>
          <h2>What can we help with?</h2>

          <div className={style['help-item']}>
            <div className={style['help-item-image']}>
              <Image
                alt='person with hair loss'
                src='/images/person1.png'
                width='370px'
                height='445px'
              />
            </div>
            <div className={style['help-item-text']}>
              <p className={style['help-item-title']}>Hair loss</p>
              <h3>Hair loss needn’t be irreversible. We can help! </h3>
              <p className={style['help-item-desc']}>
                We’re working around the clock to bring you a holistic approach
                to your wellness. From top to bottom, inside and out.
              </p>
              <p className={style['bg-text']}>01</p>
            </div>
          </div>

          <div className={`${style['help-item']} ${style['row-reverse']}`}>
            <div
              className={`${style['help-item-image']} ${style['image-reverse']}`}
            >
              <Image
                alt='an image showing a person'
                src='/images/person2.png'
                width='370px'
                height='445px'
              />
            </div>
            <div className={style['help-item-text']}>
              <p className={style['help-item-title']}>Erecetile dysfunction</p>
              <h3>
                Erections can be a tricky thing. But no need to feel down!
              </h3>
              <p className={style['help-item-desc']}>
                We’re working around the clock to bring you a holistic approach
                to your wellness. From top to bottom, inside and out.
              </p>
              <p className={`${style['bg-text']} ${style['bg-text-reverse']}`}>
                02
              </p>
            </div>
          </div>
        </section>
      </main>
    </div>
  );
};

export default HairLossPage;
