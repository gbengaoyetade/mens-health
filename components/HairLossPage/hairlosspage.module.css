.hero-section {
  width: 100%;
  height: 700px;
  background-image: url('/images/bg.png');
  background-repeat: no-repeat;
  background-size: cover;
  display: flex;
  flex-direction: column;
  padding: 1rem 8rem;
}

.head-text {
  width: 435px;
  max-width: 90vw;
  margin: auto 0;
}

.hero-paragraph {
  color: #0b3b3c;
  font-size: 18px;
  line-height: 2.2rem;
}

.head-text h1 {
  font-size: 90px;
  color: #0b3b3c;
  margin: 1rem 0;
}

.main {
  max-width: 1680px;
  width: 100vw;
  margin: 0 auto;
}

.btn {
  width: 156px;
  height: 45px;
  background: #7e0707;
  padding: 15px;
  border: none;
  color: #fff;
  font-size: 0.8rem;
}

.btn:hover {
  background: #9b0b0b;
  transition: background 0.8s;
}

.help-section {
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 0 8rem;
  background-color: #fff;
}

.help-item {
  display: flex;
  width: 65%;
  margin: 0 auto 5rem auto;
}

.help-item-image {
  margin-right: 150px;
  z-index: 1;
}

.image-reverse {
  margin-left: 150px;
  margin-right: 0;
}

.help-section h2 {
  width: 100%;
  color: #0b3b3c;
  text-align: center;
  font-size: 40px;
  line-height: 60px;
}

.help-item-title {
  color: #6d8a83;
  margin: 0;
  font-size: 14px;
  z-index: 1;
}

.help-item-text {
  color: #0b3b3c;
  width: 372px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  position: relative;
}

.help-item-text h3 {
  font-size: 28px;
  margin-top: 12px;
  margin-bottom: 1rem;
  z-index: 1;
}

.help-item-desc {
  font-size: 18px;
  z-index: 1;
}

.bg-text {
  position: absolute;
  font-size: 30rem;
  color: #f3f7f4;
  z-index: 0;
  left: -230px;
  z-index: 0;
}

.bg-text-reverse {
  left: 40px;
  z-index: 0;
}

.row-reverse {
  flex-direction: row-reverse;
}

/* Mobile styles */

@media (max-width: 568px) {
  .bg-text,
  .bg-text-reverse {
    font-size: 18rem;
    left: 50%;
    transform: translate(-50%, 0);
  }

  .hero-section {
    padding: 1rem;
    height: fit-content;
  }

  .help-section {
    padding: 0;
  }

  .help-section h2 {
    font-size: 28px;
  }

  .head-text {
    display: flex;
    flex-direction: column;
    padding: 1rem;
  }

  .head-text h1 {
    font-size: 60px;
  }

  .help-item {
    margin: 0;
    width: 100%;
    flex-direction: column;
  }

  .help-item-image,
  .image-reverse {
    margin: 0 auto;
    width: 300px;
    height: 400px;
  }

  .help-item-text {
    width: 90%;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 0 auto 18px auto;
  }

  .help-item-text h3 {
    font-size: 26px;
  }
}

@media (max-width: 320px) {
  .bg-text,
  .bg-text-reverse {
    font-size: 16em;
  }
}

@media (min-width: 569px) and (max-width: 1024px) {
  .help-item-image,
  .image-reverse {
    margin: 0 auto;
    width: 260px;
    height: auto;
  }

  .bg-text {
    font-size: 14em;
    left: -45px;
  }

  .bg-text-reverse {
    left: 45px;
  }

  .help-item {
    width: 100%;
  }

  .bg-text {
    font-size: 22em;
    left: -100px;
  }

  .bg-text-reverse {
    left: 95px;
  }
}

@media (min-width: 569px) and (max-width: 812px) and (orientation: landscape) {
  .help-item-image,
  .image-reverse {
    margin: 0 auto;
    width: 260px;
    height: auto;
  }

  .help-item-image {
    margin-right: 20px;
  }

  .image-reverse {
    margin-left: 20px;
  }

  .bg-text {
    font-size: 14em;
    left: -45px;
  }

  .bg-text-reverse {
    left: 45px;
  }

  .help-item-desc {
    font-size: 14px;
  }

  .help-item-text h3 {
    font-size: 20px;
  }
}
