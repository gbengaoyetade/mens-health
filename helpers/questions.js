export const getQuestions = () => {
  const questions = [
    {
      question: 'Which image best matches your hair loss?',
      type: 'ChoiceType',
      options: [
        {
          display:
            '<img alt="Temples" src="https://s3-eu-west-1.amazonaws.com/manualco/questions/temples-hairloss.png" srcset="https://s3-eu-west-1.amazonaws.com/manualco/questions/temples-hairloss%402x.png 2x" />',
          value: 'Temples',
          isRejection: false,
        },
        {
          display:
            '<img alt="Temples & Crown" src="https://s3-eu-west-1.amazonaws.com/manualco/questions/templescrown-hairloss.png" srcset="https://s3-eu-west-1.amazonaws.com/manualco/questions/templescrown-hairloss%402 x.png 2x"/>',
          value: 'Temples & Crown',
          isRejection: false,
        },
        {
          display:
            '<img alt="Patchy" src="https://s3-eu-west-1.amazonaws.com/manualco/questions/patchy-hairloss.png" srcset="https://s3-eu-west-1.amazonaws.com/manualco/questions/patchy-hairloss%402x.png 2x"/>',
          value: 'Patchy',
          isRejection: true,
        },
        {
          display:
            '<img alt="Moderate" src="https://s3-eu-west-1.amazonaws.com/manualco/questions/moderate-hairloss.png" srcset="https://s3-eu-west-1.amazonaws.com/manualco/questions/moderate-hairloss%402x.pn g 2x" />',
          value: 'Moderate',
          isRejection: false,
        },
        {
          display:
            '<img alt="Extensive" src="https://s3-eu-west-1.amazonaws.com/manualco/questions/extensive-hairloss.png" srcset="https://s3-eu-west-1.amazonaws.com/manualco/questions/extensive-hairloss%402x.pn g 2x"/>',
          value: 'Extensive',
          isRejection: true,
        },
        {
          display:
            '<img alt="Complete" src="https://s3-eu-west-1.amazonaws.com/manualco/questions/complete-hairloss.png" srcset="https://s3-eu-west-1.amazonaws.com/manualco/questions/complete-hairloss%402x.pn g 2x" />',
          value: 'Complete',
          isRejection: true,
        },
      ],
    },
    {
      question:
        'Have you ever been diagnosed with prostate cancer, or are you currently undergoing PSA/Prostate monitoring?',
      type: 'ChoiceType',
      options: [
        {
          display: 'Yes',
          value: true,
          isRejection: true,
        },
        {
          display: 'No',
          value: false,
          isRejection: false,
        },
      ],
    },
    {
      question:
        'Have you ever been diagnosed with breast cancer or noticed any changes in your breast tissue such as lumps, pain, nipple discharge or swelling?',
      type: 'ChoiceType',
      options: [
        {
          display: 'Yes',
          value: true,
          isRejection: true,
        },
        {
          display: 'No',
          value: false,
          isRejection: false,
        },
      ],
    },
  ];
  return questions;
};
