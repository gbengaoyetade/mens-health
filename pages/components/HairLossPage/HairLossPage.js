import style from './hairlosspage.module.css';
import Image from 'next/image';
import Footer from '../Footer';

const HairLossPage = () => {
  return (
    <>
      <main className={style.main}>
        <section className={style['hero-section']}>
          <div>
            <Image src='/images/symbol.svg' width='40px' height='38px' />
          </div>

          <div className={style['head-text']}>
            <h1>Be good to yourself</h1>
            <p className={style['hero-paragraph']}>
              We’re working around the clock to bring you a holistic approach to
              your wellness. From top to bottom, inside and out.
            </p>
            <button className={style.btn}>TAKE THE QUIZ</button>
          </div>
        </section>

        <section className={style['help-section']}>
          <h2>What can we help with?</h2>

          <div className={style['help-item']}>
            <div className={style['help-item-image']}>
              <Image src='/images/person1.png' width='370px' height='445px' />
            </div>
            <div className={style['help-item-text']}>
              <p className={style['help-item-title']}>Hair loss</p>
              <h3>Hair loss needn’t be irreversible. We can help! </h3>
              <p className={style['help-item-desc']}>
                We’re working around the clock to bring you a holistic approach
                to your wellness. From top to bottom, inside and out.
              </p>
              <p className={style['bg-text']}>01</p>
            </div>
          </div>

          <div className={`${style['help-item']} ${style['row-reverse']}`}>
            <div
              className={`${style['help-item-image']} ${style['image-reverse']}`}
            >
              <Image src='/images/person2.png' width='370px' height='445px' />
            </div>
            <div className={style['help-item-text']}>
              <p className={style['help-item-title']}>Erecetile dysfunction</p>
              <h3>
                Erections can be a tricky thing. But no need to feel down!
              </h3>
              <p className={style['help-item-desc']}>
                We’re working around the clock to bring you a holistic approach
                to your wellness. From top to bottom, inside and out.
              </p>
              <p className={`${style['bg-text']} ${style['bg-text-reverse']}`}>
                02
              </p>
            </div>
          </div>
        </section>
      </main>
      <Footer />
    </>
  );
};

export default HairLossPage;
