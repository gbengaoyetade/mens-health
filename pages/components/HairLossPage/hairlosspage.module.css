.hero-section {
  width: 100%;
  height: 700px;
  background-image: url('/images/bg.png');
  background-repeat: no-repeat;
  background-size: cover;
  display: flex;
  flex-direction: column;
  padding: 1rem 8rem;
}

.head-text {
  width: 435px;
  max-width: 90vw;
  margin: auto 0;
}

.hero-paragraph {
  color: #0b3b3c;
  font-size: 18px;
  line-height: 2.2rem;
}

.head-text h1 {
  font-size: 90px;
  color: #0b3b3c;
  margin: 1rem 0;
}

.main {
  max-width: 1680px;
  width: 100vw;
  margin: 0 auto;
}

.btn {
  width: 156px;
  height: 45px;
  background: #7e0707;
  padding: 15px;
  border: none;
  color: #fff;
  font-size: 0.8rem;
}

.btn:hover {
  background: #9b0b0b;
  transition: background 0.8s;
}

.help-section {
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 0 8rem;
  background-color: #fff;
}

.help-item {
  display: flex;
  width: 70%;
  margin: 0 auto 5rem auto;
}

.help-item-image {
  margin-right: 150px;
  z-index: 1;
}

.image-reverse {
  margin-left: 150px;
}

.help-section h2 {
  width: 100%;
  color: #0b3b3c;
  text-align: center;
  font-size: 40px;
  line-height: 60px;
}

.help-item-title {
  color: #6d8a83;
  margin: 0;
  font-size: 14px;
  z-index: 1;
}

.help-item-text {
  color: #0b3b3c;
  width: 372px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  position: relative;
}

.help-item-text h3 {
  font-size: 28px;
  margin-top: 12px;
  margin-bottom: 1rem;
  z-index: 1;
}

.help-item-desc {
  font-size: 18px;
  z-index: 1;
}

.bg-text {
  position: absolute;
  font-size: 30rem;
  color: #f3f7f4;
  z-index: 0;
  left: -230px;
  z-index: 0;
}

.bg-text-reverse {
  left: 40px;
  z-index: 0;
}

.row-reverse {
  flex-direction: row-reverse;
}
