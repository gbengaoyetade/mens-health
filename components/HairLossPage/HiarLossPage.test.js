import { render, fireEvent } from '@testing-library/react';
import HairLossPage from './HairLossPage';
import { getQuestions } from '../../helpers/questions';

const questions = getQuestions();

describe('HairLossPage', () => {
  it('should match snapshots', () => {
    const { container } = render(<HairLossPage questions={questions} />);

    expect(container).toMatchSnapshot();
  });

  it('should show quiz component when the Take The Quiz button is clicked', () => {
    const { getByText } = render(<HairLossPage questions={questions} />);

    fireEvent.click(getByText('TAKE THE QUIZ'));

    expect(
      getByText('Which image best matches your hair loss?')
    ).toBeInTheDocument();
  });

  it('should show back button when not answering the first question', () => {
    const { getByText, getByTestId } = render(
      <HairLossPage questions={questions} />
    );

    fireEvent.click(getByText('TAKE THE QUIZ'));
    fireEvent.click(getByTestId('0_option_0'));

    expect(getByTestId('back-btn')).toBeInTheDocument();
  });

  it('should show previous question when the back button is clicked', () => {
    const { getByText, getByTestId } = render(
      <HairLossPage questions={questions} />
    );

    fireEvent.click(getByText('TAKE THE QUIZ'));
    fireEvent.click(getByTestId('0_option_0'));
    fireEvent.click(getByTestId('back-btn'));

    expect(
      getByText('Which image best matches your hair loss?')
    ).toBeInTheDocument();
  });
});
