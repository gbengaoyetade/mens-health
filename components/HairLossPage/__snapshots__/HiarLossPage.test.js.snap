// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HairLossPage should match snapshots 1`] = `
<div>
  <div
    class="main"
  >
    <main>
      <section
        class="hero-section"
      >
        <div>
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAiIGhlaWdodD0iMzgiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="app logo"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="head-text"
        >
          <h1>
            Be good to yourself
          </h1>
          <p
            class="hero-paragraph"
          >
            We’re working around the clock to bring you a holistic approach to your wellness. From top to bottom, inside and out.
          </p>
          <button
            class="btn"
          >
            TAKE THE QUIZ
          </button>
        </div>
      </section>
      <section
        class="help-section"
      >
        <h2>
          What can we help with?
        </h2>
        <div
          class="help-item"
        >
          <div
            class="help-item-image"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzcwIiBoZWlnaHQ9IjQ0NSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="person with hair loss"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="help-item-text"
          >
            <p
              class="help-item-title"
            >
              Hair loss
            </p>
            <h3>
              Hair loss needn’t be irreversible. We can help! 
            </h3>
            <p
              class="help-item-desc"
            >
              We’re working around the clock to bring you a holistic approach to your wellness. From top to bottom, inside and out.
            </p>
            <p
              class="bg-text"
            >
              01
            </p>
          </div>
        </div>
        <div
          class="help-item row-reverse"
        >
          <div
            class="help-item-image image-reverse"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzcwIiBoZWlnaHQ9IjQ0NSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="an image showing a person"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="help-item-text"
          >
            <p
              class="help-item-title"
            >
              Erecetile dysfunction
            </p>
            <h3>
              Erections can be a tricky thing. But no need to feel down!
            </h3>
            <p
              class="help-item-desc"
            >
              We’re working around the clock to bring you a holistic approach to your wellness. From top to bottom, inside and out.
            </p>
            <p
              class="bg-text bg-text-reverse"
            >
              02
            </p>
          </div>
        </div>
      </section>
    </main>
  </div>
</div>
`;
